src/robot/running/arguments/embedded.py

```diff
@@ -53,10 +54,8 @@ def parse(self, string):
         return name, args
 
     def _get_name_and_pattern(self, name):
-        if ':' not in name:
-            return name, self._default_pattern
-        name, pattern = name.split(':', 1)
-        return name, self._format_custom_regexp(pattern)
+        name, type_, pattern = split_variable_name_without_bracket_and_type_and_pattern(name)
+        return ':'.join(e for e in [name, type_] if e), self._format_custom_regexp(pattern) if pattern else self._default_pattern
```

Comment:
Didn't realize earlier that handling types with embedded arguments is a bit complicated. They already support `arg:pattern` syntax and we cannot easily differentiate `arg:type` or `arg:pattern:type` syntax from that. This is especially problematic because the `pattern` may also contain literal `:` character.

I'm not sure what's the best way to solve this issue. What I do know is that splitting this PR into smaller parts makes discussing about details like this a lot easier.

Reply:
I split the PR after you request, this part of code is not anymore